services/web/dev.Dockerfile

```diff
@@ -1,30 +1,46 @@
 # Create a new build stage from a base image
-FROM node:23-alpine AS builder
+FROM node:23-alpine AS build-stage
+
+# https://nodejs.org/en/learn/getting-started/nodejs-the-difference-between-development-and-production
+# Use production node environment by default.
+ENV NODE_ENV=production
 
 # Change the working directory to /app inside of the container. If that directory does not already exist, it will
 # be created. Reference: https://docs.docker.com/reference/dockerfile/#workdir
 WORKDIR /app
 
-# Copy first only the package.json and package-lock.json files to leverage Docker cache
-COPY package*.json ./
-
-# Install dependencies (npm ci makes sure the exact versions in the lockfile gets installed)
-RUN npm ci
+# Download dependencies as a separate step to take advantage of Docker's caching.
+# Leverage a cache mount to /root/.npm to speed up subsequent builds.
+# Leverage a bind mounts to package.json and package-lock.json to avoid having to copy them into
+# into this layer.
+# See https://docs.docker.com/build/building/best-practices/#add-or-copy
+# We use npm ci to do a clean install of the dependencies. https://docs.npmjs.com/cli/v11/commands/npm-ci
+RUN --mount=type=bind,source=package.json,target=package.json \
+    --mount=type=bind,source=package-lock.json,target=package-lock.json \
+    --mount=type=cache,target=/root/.npm \
+    npm ci
 
 # Copy the rest of the application files
 COPY . .
 
-# The environment variables set using ENV will persist when a container is run from the resulting image.
 ENV VITE_ENV="DEV"
 
 # Build the app
 RUN npm run build
 
 # Bundle static assets with nginx
-FROM nginx:latest AS dev
+# TODO: Change image with https://hub.docker.com/r/nginxinc/nginx-unprivileged when the auto renew of ssl certs is
+# implemented, as oer: https://docs.docker.com/build/building/best-practices/#user
+FROM nginx:1.27-alpine AS dev
```

Comment:
Why do you need a dev dockerfile?

Reply:
cause we are copying a different nginx configs depending on the env, and I didn't want to do if-else copy. (COPY does not allow it, and I had to use RUN)

Reply:
I'm not sure I understand what you mean by exposing too many details. Changes in directory structure would always mean editing the path pointing to the configuration - no matter whether it was defined as a build argument or hardcoded. 

I doubt that your autocomplete extension is firing off errors if path is incorrect. Hence, having the path hardcoded there is not of any net benefit.

You could possibly introduce a linter to catch these errors but that's an overkill. Of course that changes in structure would dictate changes in configurations. But do you really expect too much of a change in structure here?

Reply:
Tbh, not really. So, you recommend either hardcoding it in the CD pipeline or extracting it as a GitHub ENV var, right? I prefer the latter if that's the case.

Reply:
I recommend conditionally inferring it in the deployment pipeline

If (branch === production) prefix = "prod." else "";
arg PATH_TO_NGINX_FILE = path + prefix + file

Reply:
@asynchroza Fixed in [58d6367](https://github.com/AUBGTheHUB/monolith/pull/995/commits/58d63675609ce780d6fe3df3f761351744fb2734)